### PR TITLE
Improve log messages

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#24368|

## Description 

This PR improves the log messages by adding more information and fixing grammar and punctuation rules.

### CODE REVIEW

File: src/shared_modules/utils/socketWrapper.hpp [hunk 1]
```diff
@@ -685,7 +685,7 @@ class Socket final : public T
                 if (ret <= 0)
                 {
                     m_unsentPacketList.emplace(m_sendDataBuffer.data() + amountSent, bufferSize - amountSent);
-                    throw std::runtime_error {"Error sending data to socket: " + std::to_string(errno) + " " +
+                    throw std::runtime_error {"Error sending data to socket: " + std::string(std::strerror(errno)) + " " +
                                               std::string(strerror(errno))};
```
Comment (sebasfalcone) [inline src/shared_modules/utils/socketWrapper.hpp:689]: Repeated error string
```suggestion
                    throw std::runtime_error {"Error sending data to socket: " + std::string(std::strerror(errno))};
```

File: src/wazuh_modules/vulnerability_scanner/qa/test_efficacy_log.py [hunk 2]
```diff
@@ -157,98 +157,94 @@ def run_process_and_monitor_log(request, run_on_end):
...
-            if timeout < 8*basetimeout:
-                tail_log(log_file, expected_lines, found_lines, timeout)
-                timeout = 1.5*timeout
-            else:
-                LOGGER.error(f"Timeout waiting for log line: {expected_line}")
-                timeout = basetimeout
-                break
-
-    process.terminate()
+    with subprocess.Popen(command) as process:
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/qa/test_efficacy_log.py:160]: A little bit out of scope, but the changes below are just the indentation added

This improves the management of the process and prevents leaks for not calling terminate

File: src/wazuh_modules/vulnerability_scanner/qa/test_efficacy_log.py [hunk 3]
```diff
@@ -157,98 +157,94 @@ def run_process_and_monitor_log(request, run_on_end):
...
+                LOGGER.debug(f"Waiting for log line: {expected_line}")
+                if (timeout < 60 and test_folder.name == '000') or (timeout < 5 and not test_folder.name == '000'):
+                    tail_log(log_file, expected_lines, found_lines, timeout)
+                    timeout = 2*timeout
+                else:
+                    LOGGER.error(f"Timeout waiting for log line: {expected_line}")
+                    timeout = 1
+                    break
+
+        process.terminate()
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/qa/test_efficacy_log.py:247]: Note, this is not necessary due to "with subprocess.Popen(command) as process:"

**Review by sebasfalcone — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/vulnerability_scanner/qa/test_efficacy_log.py (hunk 2 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/qa/test_efficacy_log.py:160]: Indeed out of scope, it was useful while running the tests, but this is present in 4.8.0

**Review by sebasfalcone — APPROVED** [review]

**Pull request merged**